core/util.py

```diff
@@ -65,3 +65,20 @@ def get_project_number(project_id):
     :return: project number
     """
     return int(project_id.split("-")[2])
+
+
+def construct_index(section: str, column: str, rows: tuple[int] | int) -> str:
+    """Constructs the index of an error from the column and rows it occurred in.
+
+    :param section: the internal table name where the error occurred
+    :param column: the internal table name where the error occurred
+    :param rows: tuple or int of row indexes where the error occurred
+    :return: indexes tuple of constructed letter and number indexes
+    """
+
+    column_letter = TABLE_AND_COLUMN_TO_ORIGINAL_COLUMN_LETTER[section][column]
+    if type(rows) is int:
+        indexes = column_letter + str(rows)
+    else:
+        indexes = ", ".join([column_letter + str(row) for row in rows])
```

Comment:
We shouldn't use arrays in order to store the indexes. These unit tests currently pass because all of the arrays only contain one value, but if they contain more than two the tests will fail. You can read more about it here: https://stackoverflow.com/questions/10062954/valueerror-the-truth-value-of-an-array-with-more-than-one-element-is-ambiguous

We can use tuples instead, which don't encounter this issue

Reply:
To push back on this a little, I thought a good rule of thumb is to use a python list for homogeneous data and since the indexes will all be ints it seems to make sense.

Also I can add more than one value to the index lists and still have them pass. See updated test_validate_locations_failure.

The link you sent seems to be talking more about numpy arrays. Whereas python lists don't have the same behaviour I belive. Maybe needs more thought we should choose one, list or tuple and stick with that.

Reply:
I raised this because for the non-R4 specific validation tests, including indexes as a list was having tests fails with a ValueError when I tried to compare them. The issue seemed to be that my arrays were being treated as numpy arrays, and I thought this might end up being the case with all of our arrays of indexes. 

Investigating, it looks like this comes from lines of code like:
```
failed_indexes = sheet.index[(pd.isna(sheet[column]) | (sheet[column].astype("string") == ""))]
```
but thinking about it, the lists can be converted from numpy arrays to Python arrays with tolist(). 

Provided we've got more than one value for indexes (where we expect multiple) in the tests then I don't think it matters. This PR precedes mine so we can go with lists :)

Reply:
Okay, no problem. I'll go with lists then for this case and try and address the other comments here so that we can get thisone in soon